tests/tornado/test_tchannel.py

```diff
@@ -88,3 +89,23 @@ def test_advertise_should_listen_if_not_called_yet(tchannel, mock_server):
     )
 
     assert tchannel.is_listening() is True
+
+
+def test_advertise_should_take_a_router_file(tchannel, mock_server):
+
+    mock_server.expect_call('ad', 'json').and_write(
```

Comment:
mock_server is kinda lame and we should avoid introducing new uses. Check out `tests/test_tchannel.py` to see how to write a server inline.